download-view/src/main/java/cm/aptoide/pt/download_view/presentation/DownloadViewModel.kt

```diff
@@ -208,5 +223,19 @@ class DownloadViewModel(
     }
   }
 
+  private fun migrate(resolver: ConstraintsResolver) {
+    try {
+      uninstall()
+      appInstaller.packageInfoFlow.filter { info ->
+        info == null
+      }
+        .map {
+          install(resolver)
```

Comment:
Do we a business decision about UX if download or installation will fail at this moment? 🤔

Reply:
Do we need it ? It will be handled by the "normal flow". Here we just enqeue the installation task. It will be handled normally by the flow above.
Only problem would be if there was an error while enqueing that task but then it would stay on the install button, which is what actaully makes sense

Reply:
The problem is that in this case user actually already uninstalled his game with all his progress 😅